bundles/org.openhab.binding.chatgpt/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -19,6 +19,26 @@
 				<label>API Key</label>
 				<description>API key to access the account</description>
 			</parameter>
+			<parameter name="apiUrl" type="text" required="false">
+				<label>API URL</label>
+				<description>The server API where to reach the AI service.</description>
+				<default>https://api.openai.com/v1/chat/completions</default>
+				<advanced>true</advanced>
+				<options>
+					<option value="https://api.openai.com/v1/chat/completions">https://api.openai.com/v1/chat/completions</option>
+				</options>
+				<limitToOptions>false</limitToOptions>
+			</parameter>
+			<parameter name="modelUrl" type="text" required="false">
+				<label>Base URL</label>
+				<description>The model url where to retrieve the available models from.</description>
+				<default>https://api.openai.com/v1/models</default>
+				<advanced>true</advanced>
+				<options>
+					<option value="https://api.openai.com/v1/models">https://api.openai.com/v1/models</option>
```

Comment:
@mudler Is such a url also available from LocalAI, so that the available models can be retrieved? If so, what's the correct url?

Reply:
yep it does exist, it would be equivalent to:

http://localhost:8080/v1/models

Reply:
the text generation endpoints are documented here : https://localai.io/features/text-generation/ - however there is 1:1 compatibility with OpenAI, so besides the URL and the model there is nothing else to touch! :)

Reply:
Ok, great.
The tricky part might be https://github.com/openhab/openhab-addons/blob/3cd8c3cf7872c30f9b60d72793383758ea055b08/bundles/org.openhab.binding.chatgpt/src/main/resources/OH-INF/thing/thing-types.xml#L55 - we are currently setting the model parameter to `gpt-3.5-turbo` as a default, which won't exist for a LocalAI installation. If users do not adapt this parameter: How will LocalAI react? Will it result in an error when this model is requested or will it silently fall back to another default model?

Reply:
by default if no model is specified in the request, or if there is only one installed it will redirect all the requests to it - however in LocalAI the user is free to call models as they like, so they can mimick gpt-4 behind the scene